src/fw/drivers/nrf5/qspi.c

```diff
@@ -1,138 +1,207 @@
-#include "drivers/qspi.h"
+/*
+ * Copyright 2025 Core Devices LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdint.h>
 
 #include "board/board.h"
-#include "drivers/dma.h"
-#include "drivers/flash/flash_impl.h"
+#include "drivers/qspi.h"
 #include "drivers/flash/qspi_flash.h"
-#include "drivers/flash/qspi_flash_definitions.h"
-#include "drivers/gpio.h"
-#include "drivers/periph_config.h"
+#include "drivers/flash/flash_impl.h"
 #include "drivers/qspi_definitions.h"
 #include "flash_region/flash_region.h"
 #include "kernel/util/delay.h"
 #include "kernel/util/sleep.h"
-#include "kernel/util/stop.h"
-#include "mcu/cache.h"
-#include "system/logging.h"
 #include "system/passert.h"
 #include "util/math.h"
 
-#define NRF5_COMPATIBLE
+#define CMSIS_COMPATIBLE
```

Comment:
hm?

Reply:
we just need CMSIS APIs from mcu.h (mcu.h should really disappear)